flight_backend/api/models/flight.py

```diff
@@ -0,0 +1,69 @@
+import uuid
+from django.db import models
+from djmoney.models.fields import MoneyField
+from django.utils.translation import ugettext_lazy as _
+from .. import constants
+
+
+class Aircraft(models.Model):
+    name = models.CharField(max_length=50)
+    manufacturer = models.CharField(max_length=50)
+    model = models.CharField(max_length=50)
+    capacity = models.PositiveSmallIntegerField(default=500)
+
+    def __str__(self):
+        return f"{self.manufacturer} {self.model}"
+
+
+class Location(models.Model):
+    country = models.CharField(max_length=50)
+    country_code = models.CharField(max_length=4)
+    city = models.CharField(max_length=50)
+    latitude = models.DecimalField(max_digits=9, decimal_places=6)
+    longitude = models.DecimalField(max_digits=9, decimal_places=6)
+
+    def __str__(self):
+        return f"{self.city}, {self.country}"
+
+
+class Flight(models.Model):
+    """
+    This model defines a Flight object
+    """
+    flight_number = models.CharField(_('Enter flight number'), max_length=50, blank=True, unique=True)
+    departure_date = models.DateField(_(u"Departure Date"), blank=True, null=True)
+    arrival_date = models.DateField(_(u"Arrival Date"), blank=True, null=True)
+    departure = models.ForeignKey("Location", on_delete=models.SET_NULL, null=True, related_name='departure')
+    destination = models.ForeignKey("Location", on_delete=models.SET_NULL, null=True, related_name='destination')
+    flight_duration = models.DurationField(blank=True)
+    economy_price = MoneyField(max_digits=14, decimal_places=2, default_currency='USD', null=True, blank=True)
+    business_price = MoneyField(max_digits=14, decimal_places=2, default_currency='USD', null=True, blank=True)
+    aircraft = models.ForeignKey("Aircraft", on_delete=models.SET_NULL, null=True)
+    status = models.CharField(max_length=50, choices=constants.FLIGHT_STATUS, default=constants.AVAILABLE)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return "{}".format(self.flight_number)
+
+
+class Ticket(models.Model):
+    """
+    Model defines a Ticket object
+    """
+    user = models.ForeignKey("User", on_delete=models.CASCADE)
+    flight = models.ForeignKey("Flight", on_delete=models.CASCADE)
+    ticket_id = models.CharField(max_length=50, blank=True)
+    ticket_class = models.CharField(max_length=10, choices=constants.CLASS)
+    phone_number = models.CharField(max_length=50)
+    passport_number = models.CharField(max_length=50)
+    contact_address = models.CharField(max_length=255)
```

Comment:
i think the contact details should be part of the user data

Reply:
ok, I will make the changes.